mk/waf.mk

```diff
@@ -95,12 +95,19 @@ all:
 	@echo BINARY_NAME=$(BINARY_NAME)$(BINARY_FRAME)
 	@echo BINARY_EXTENSION=$(BINARY_EXTENSION)
 	@echo BINARY_EXTENSION_DEST=$(BINARY_EXTENSION_DEST)
+	mkdir -p $(SKETCHBOOK)/build
 	cd $(SKETCHBOOK) && modules/waf/waf-light --board $(TARGET_BOARD) configure
 	cd $(SKETCHBOOK) && modules/waf/waf-light $(TARGET_BUILD) $(WAF_FLAGS)
 	@cp $(SKETCHBOOK)/build/$(BIN_DIR)/bin/$(BINARY_NAME)$(BINARY_FRAME)$(BINARY_EXTENSION) $(SKETCH)$(BINARY_EXTENSION_DEST)
 	@ls -l $(SKETCH)$(BINARY_EXTENSION_DEST)
 	@echo "Build done $(SKETCH)$(BINARY_EXTENSION_DEST)"
 
 clean:
-	@cd $(SKETCHBOOK) && modules/waf/waf-light clean
+	mkdir -p $(SKETCHBOOK)/build
+	@cd $(SKETCHBOOK) && modules/waf/waf-light --board $(TARGET_BOARD) configure clean
+
+
+px4-clean:
+	@echo Removing px4 build directories
+	@rm -rf $(SKETCHBOOK)/build/px4*
```

Comment:
This is because clean in px4 boards isn't cleaning everything? If so, we should fix it in waf.